bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BXMLItem.java

```diff
@@ -890,4 +891,42 @@ public synchronized void attemptFreeze(BVM.FreezeStatus freezeStatus) {
             this.freezeStatus = freezeStatus;
         }
     }
+
+    private static class BXmlAttrMap extends BMap {
+        private final BXMLItem bXmlItem;
+        private boolean constructed = false;
+
+        BXmlAttrMap(BXMLItem bXmlItem) {
+            super(new BMapType(BTypes.typeString));
+            this.bXmlItem = bXmlItem;
+        }
+
+        void finishConstruction() {
+            constructed = true;
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public void put(Object key, BValue value) {
+            super.put(key, value);
+            if (constructed) {
+                setAttribute((String) key, value.stringValue());
+            }
+        }
+
+        private void setAttribute(String key, String value) {
+            String url = null;
+            String localName = key;
+
+            int endOfUrl = key.indexOf('}');
```

Comment:
What happens if the namespace contains braces? For example,
```ballerina
xmlns "foo}}bar" as foobar;
```
Then the `key` will be `{foo}}bar}age` and `url` will be `foo`.